File: UnitConverter/app/src/main/java/com/example/unitconverter/MainActivity.java
package com.example.unitconverter;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    private Button button;
    private TextView txtView;
    private EditText editText;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        EditText ed = (EditText) findViewById(R.id.edt1);
        TextView tv = (TextView) findViewById(R.id.txt2);
        Button bb1 = (Button) findViewById(R.id.b1);
        bb1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String s = ed.getText().toString();
                int kg = Integer.parseInt(s);
                double pound = 2.205 * kg;
                tv.setText("The corresponding value of pound is:" + pound);
            }
        });

    }
}

File: MemeShare/app/src/main/java/com/example/memeshare/MainActivity.kt
package com.example.memeshare

import android.content.Intent
import android.graphics.drawable.Drawable
import android.os.Bundle
import android.view.View
import android.view.WindowManager
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.android.volley.Request
import com.android.volley.toolbox.JsonObjectRequest
import com.android.volley.toolbox.Volley
import com.bumptech.glide.Glide
import com.bumptech.glide.load.DataSource
import com.bumptech.glide.load.engine.GlideException
import com.bumptech.glide.request.RequestListener
import com.bumptech.glide.request.target.Target
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    var currentimageURL:String?=null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        supportActionBar!!.hide()
        window.setFlags(
            WindowManager.LayoutParams.FLAG_FULLSCREEN,
            WindowManager.LayoutParams.FLAG_FULLSCREEN
        )
        loadMeme()
    }
    private fun loadMeme(){
        // Instantiate the RequestQueue.
        progressOccur.visibility=View.VISIBLE
        val queue = Volley.newRequestQueue(this)
        val url = " https://meme-api.herokuapp.com/gimme"

// Request a string response from the provided URL.
        val jsonObjectRequest = JsonObjectRequest(
            Request.Method.GET, url,null,
            { response ->
                currentimageURL=response.getString("url")
                Glide.with(this).load(currentimageURL).listener(object:RequestListener<Drawable>{
                    override fun onLoadFailed(e: GlideException?, model: Any?, target: Target<Drawable>?, isFirstResource: Boolean): Boolean {
                        progressOccur.visibility=View.GONE
                        return false
                    }

                    override fun onResourceReady(resource: Drawable?, model: Any?, target: Target<Drawable>?, dataSource: DataSource?, isFirstResource: Boolean): Boolean {
                        progressOccur.visibility=View.GONE
                        return false
                    }
                }).into(memeImage)


            },
            {
                Toast.makeText(this,"Something went Wrong", Toast.LENGTH_LONG).show()
            })

// Add the request to the RequestQueue.
        queue.add(jsonObjectRequest)
    }

    fun nextMeme(view: View) {
        loadMeme()
//        Toast.makeText(this,"Next Mene will be Shown", Toast.LENGTH_LONG).show()
    }
    fun shareMeme(view: View) {
            val intent=Intent(Intent.ACTION_SEND)
            intent.type="text/plain"
            intent.putExtra(Intent.EXTRA_TEXT,"Hey ,Checkout this cool Meme $currentimageURL")
            val chooser=Intent.createChooser(intent,"Share this meme")
            startActivity(chooser)
//        Toast.makeText(this,"Meme is getting shared", Toast.LENGTH_LONG).show()
    }
}